ext/openssl/openssl.c

```diff
@@ -458,21 +458,113 @@ void php_openssl_store_errors()
 }
 /* }}} */
 
-static int ssl_stream_data_index;
+/* {{{ openssl file path check error function */
+static void php_openssl_check_path_error(uint32_t arg_num, int type, const char *format, ...)
+{
+	va_list va;
+	char *message = NULL;
+	const char *arg_name;
 
+	va_start(va, format);
+
+	if (type == E_ERROR) {
+		zend_vspprintf(&message, 0, format, va);
+		zend_argument_value_error(arg_num, message);
+		efree(message);
+	} else {
+		arg_name = get_active_function_arg_name(arg_num);
+		php_verror(NULL, arg_name, type, format, va);
+	}
+	va_end(va);
+}
 /* }}} */
 
-/* {{{ openssl open_basedir check */
-inline static int php_openssl_open_base_dir_chk(char *filename)
+/* {{{ openssl file path check extended */
+static bool php_openssl_check_path_ex(
+		const char *file_path, size_t file_path_len, char *real_path, uint32_t arg_num,
+		bool contains_file_protocol, bool is_from_array, const char *option_name)
 {
-	if (php_check_open_basedir(filename)) {
-		return -1;
+	const char *fs_file_path;
+	size_t fs_file_path_len;
+	const char *error_msg = NULL;
+	int error_type = E_WARNING;
+
+	if (file_path_len == 0) {
+		real_path[0] = '\0';
+		return true;
 	}
 
-	return 0;
+	if (contains_file_protocol) {
+		size_t path_prefix_len = sizeof("file://") - 1;
+		fs_file_path = file_path + path_prefix_len;
+		fs_file_path_len = file_path_len - path_prefix_len;
+	} else {
+		fs_file_path = file_path;
+		fs_file_path_len = file_path_len;
+	}
+
+	if (CHECK_NULL_PATH(fs_file_path, fs_file_path_len)) {
+		error_msg = "must not contain any null bytes";
+		error_type = E_ERROR;
+	} else if (expand_filepath(fs_file_path, real_path) == NULL) {
+		error_msg = "The argument must be a valid file path";
+	}
+
+	if (error_msg != NULL) {
+		if (arg_num == 0) {
+			const char *option_title = option_name ? option_name : "unknown";
+			const char *option_label = is_from_array ? "array item" : "option";
+			php_error_docref(NULL, E_WARNING, "Path '%s' for %s %s %s",
+				real_path, option_title, option_label, error_msg);
+		} else if (is_from_array && option_name != NULL) {
+			php_openssl_check_path_error(
+					arg_num, error_type, "option %s array item %s", option_name, error_msg);
+		} else if (is_from_array) {
+			php_openssl_check_path_error(arg_num, error_type, "array item %s", error_msg);
+		} else if (option_name != NULL) {
+			php_openssl_check_path_error(
+					arg_num, error_type, "option %s %s", option_name, error_msg);
+		} else {
+			php_openssl_check_path_error(arg_num, error_type, error_msg);
```

Comment:
I'd use `"%s", error_msg`. It's safe here, but generally passing a variable to a format argument is a bad idea.